src/@types/Navigation.d.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  BottomTabBarProps,
+  BottomTabScreenProps,
+} from '@react-navigation/bottom-tabs';
+import { NativeStackScreenProps } from '@react-navigation/native-stack';
+
+declare global {
+  type StackParamList = {
+    Login: undefined;
+    Signup: undefined;
+    BottomTabs: undefined;
+  };
+
+  type LoginScreenParamList = NativeStackScreenProps<StackParamList, 'Login'>;
+  type SignupScreenParamList = NativeStackScreenProps<StackParamList, 'Signup'>;
+
+  type TabsParamList = {
+    Dashboard: undefined;
+    Profile: undefined;
+    EventListStack: undefined;
+  };
+
+  type DashboardScreenParamList = BottomTabScreenProps<
+    TabsParamList,
+    'Dashboard'
+  >;
+  type EventListNavigation = BottomTabScreenProps<
+    TabsParamList,
+    'EventListStack'
+  >;
+  type ProfileScreenParamList = BottomTabScreenProps<TabsParamList, 'Profile'>;
+
+  type EventStackParamList = {
+    EventList: undefined;
+    AddEvent: undefined;
```

Comment:
why add or edit screens  are different?

Reply:
just screens are different the main component that contains the form for add/edit event is the same

Reply:
Ok I will change them. This files are just for temporary routing